manager/allocator/network.go

```diff
@@ -81,6 +81,7 @@ func (a *Allocator) doNetworkInit(ctx context.Context) error {
 		unallocatedNetworks: make(map[string]*api.Network),
 		ingressNetwork:      newIngressNetwork(),
 	}
+	a.netCtx = nc
```

Comment:
Sounds kind of dangerous TBH. What about a deferred closure in `doNetworkInit` that clears `a.netCtx` if an error is being returned?